### ISSUE TITLE
pdim code is wrong for nontrivially presented 0 module!

### ISSUE THREAD

**Issue by eisenbud** [Macaulay2/M2#450]
```
S = ZZ/101[x]
pdim coker map(S^1,S^1,1)
```

returns 1 (although the module is 0)
because pdim is defined incorrectly as length res trim M 
it should be
pdim Module := M-> length res minimalPresentation M

Also:
```
f = map (S^1,S^1,1)
M = coker f
N= subquotient(f,f)
assert(M==N)
assert (res M == res N)
```
succeed; but:
```
assert (res trim M == res trim N)
```
fails!
The doc node for res should also say that the first matrix of res M

**Comment by DanGrayson** [comment]
The problem with trim is that it does less computation when the generating matrix is missing.

```
R = QQ[]
M = subquotient ( map(R^1,R^1,1), map(R^1,R^1,1) )
N = subquotient (               , map(R^1,R^1,1) )
P = subquotient ( map(R^1,R^0,0), map(R^1,R^1,1) )
M == 0
N == 0
trim M
trim N
assert (trim M === trim N)
N == 0
pdim N
assert ( pdim N == 0 )
```

**Comment by DanGrayson** [comment]
resolved in 9a051a54cc0ba9c86cb2929234f66b51034a2049
